packages/react-components/lib/tasks/task-form.tsx

```diff
@@ -311,11 +299,14 @@ export interface CreateTaskFormProps
   pickupPoints?: Record<string, string>;
   dropoffPoints?: Record<string, string>;
   favoritesTasks?: TaskFavorite[];
-  scheduleToEdit?: Schedule;
-  // requestTask is provided only when editing a schedule
-  requestTask?: TaskRequest;
-  onDispatchTask(task: TaskRequest, robotDispatchTarget: RobotDispatchTarget | null): Promise<void>;
-  onScheduleTask(task: TaskRequest, schedule: Schedule): Promise<void>;
+  schedule?: Schedule;
+  taskRequest?: TaskRequest;
+  onDispatchTask?(
+    task: TaskRequest,
+    robotDispatchTarget: RobotDispatchTarget | null,
+  ): Promise<void>;
+  onScheduleTask?(task: TaskRequest, schedule: Schedule): Promise<void>;
+  onEditScheduleTask?(task: TaskRequest, schedule: Schedule): Promise<void>;
```

Comment:
Add docs for these 2 props and `schedule`. iiuc, only `onEditScheduleTask` is used in "edit" mode, which is controlled by the `schedule` prop and `onScheduleTask` is used otherwise. They are mutually exclusive and will never be used together.

Better would be to refactor this to use the micro app hooks, the task form should be able to handle all the api calls itself and all these callbacks should be able to be replaced with a single `onClose` callback. But I leave it up to you if you want to do the refactor now.

Reply:
they technically can be used together, just that we choose not to provide a callback in whichever place we are creating a task form, i.e. the edit schedule component can also provide `onScheduleTask` but because of the function itself, we choose not to provide it.

yeah using hooks will be ideal, I will help look into it when we finally merge react components and dashboard